mongodbatlas/data_source_mongodbatlas_federated_settings_identity_providers.go

```diff
@@ -0,0 +1,391 @@
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+
+	federationSettingsID, federationSettingsIDOk := d.GetOk("federation_settings_id")
+
+	options := &matlas.ListOptions{
+		PageNum:      d.Get("page_num").(int),
+		ItemsPerPage: d.Get("items_per_page").(int),
+	}
+
+	if !federationSettingsIDOk {
+		return diag.FromErr(errors.New("either federation_settings_id must be configured"))
+	}
+
+	federatedSettingsIdentityProviders, _, err := conn.FederatedSettings.ListIdentityProviders(ctx, federationSettingsID.(string), options)
+	if err != nil {
+		return diag.Errorf("error getting federatedSettings IdentityProviders assigned (%s): %s", federationSettingsID, err)
+	}
+
+	if err := d.Set("results", flattenFederatedSettingsIdentityProvider(federatedSettingsIdentityProviders)); err != nil {
+		return diag.FromErr(fmt.Errorf("error setting `result` for federatedSettings IdentityProviders: %s", err))
+	}
+
+	d.SetId(federationSettingsID.(string))
+
+	return nil
+}
+
+func flattenFederatedSettingsIdentityProvider(federatedSettingsIdentityProvider []matlas.FederatedSettingsIdentityProvider) []map[string]interface{} {
+	var federatedSettingsIdentityProviderMap []map[string]interface{}
+
+	if len(federatedSettingsIdentityProvider) > 0 {
+		federatedSettingsIdentityProviderMap = make([]map[string]interface{}, len(federatedSettingsIdentityProvider))
+
+		for i := range federatedSettingsIdentityProvider {
+			federatedSettingsIdentityProviderMap[i] = map[string]interface{}{
+				"acs_url":                      federatedSettingsIdentityProvider[i].AcsURL,
+				"associated_domains":           federatedSettingsIdentityProvider[i].AssociatedDomains,
+				"associated_orgs":              flattenAssociatedOrgs(federatedSettingsIdentityProvider[i].AssociatedOrgs),
+				"audience_uri":                 federatedSettingsIdentityProvider[i].AudienceURI,
+				"display_name":                 federatedSettingsIdentityProvider[i].DisplayName,
+				"issuer_uri":                   federatedSettingsIdentityProvider[i].IssuerURI,
+				"okta_idp_id":                  federatedSettingsIdentityProvider[i].OktaIdpID,
+				"pem_file_info":                flattenPemFileInfo(*federatedSettingsIdentityProvider[i].PemFileInfo),
+				"request_binding":              federatedSettingsIdentityProvider[i].RequestBinding,
+				"response_signature_algorithm": federatedSettingsIdentityProvider[i].ResponseSignatureAlgorithm,
+				"sso_debug_enabled":            federatedSettingsIdentityProvider[i].SsoDebugEnabled,
+				"sso_url":                      federatedSettingsIdentityProvider[i].SsoURL,
+				"status":                       federatedSettingsIdentityProvider[i].Status,
+			}
+		}
+	}
+
+	return federatedSettingsIdentityProviderMap
+}
+
+func flattenAssociatedOrgs(associatedOrgs []*matlas.AssociatedOrgs) []map[string]interface{} {
+	var associatedOrgsMap []map[string]interface{}
+
+	if len(associatedOrgs) > 0 {
```

Comment:
```suggestion
	if len(associatedOrgs) == 0 {
	   return nil }
```
Trying to remove some nested if from this function as it is hard to read

Reply:
Fixed